Formula/smug.rb

```diff
@@ -0,0 +1,21 @@
+class Smug < Formula
+  desc "Automate your tmux workflow"
+  homepage "https://github.com/ivaaaan/smug"
+  url "https://github.com/ivaaaan/smug/archive/refs/tags/v0.2.5.tar.gz"
+  sha256 "314711f616ea71613fe883fe579e7d3bc0051f540f27c7925a1ce6a0ece69378"
+  license "MIT"
+
+  depends_on "go" => :build
+  depends_on "tmux" => :test
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}")
+  end
+
+  test do
+    (testpath/"test.yml").write <<~EOS
+      session: test
+    EOS
+    system "smug", "start", "--file", "test.yml"
```

Comment:
can we assert the result?

Reply:
The result is just non-zero exit

Reply:
I think this seems fine. Does this start a Tmux session named `test`? If so, we might want to check that such a session was created using `tmux list-sessions`. We probably also want to do `tmux kill-session` after that.

Reply:
Since the config does not define any windows, it just parses the config, and exits. It should not leave any lingering sessions.

Reply:
What does it do with the parsed config?

Reply:
I'm not sure. At a high level, I think it just maps yaml to tmux commands, and then executes them in some sort of order.

Reply:
@jedahan it will start a session anyway. @carlocab would it work if the test runs two commands?
It will start and stop sessions, which seems like enough checking 

```
smug start --file test.yml
smug stop --file test.yml
```

Reply:
Can we check there is a Tmux session running in between the two with something like `tmux list-sessions`?

Reply:
Also I pretty much am at my limit of time/ability to work on this further, so I am fine closing the PR since this is becoming much more time consuming than I originally thought.